libconnman-qt/networkmanager.cpp

```diff
@@ -911,6 +922,30 @@ void NetworkManager::propertyChanged(const QString &name, const QDBusVariant &va
     propertyChanged(name, value.variant());
 }
 
+void NetworkManager::handleTetheringClientsChanged(const QStringList &added, const QStringList &removed)
+{
+    bool change = false;
+
+    for (QStringList::ConstIterator i = added.constBegin(); i != added.constEnd(); ++i) {
+        qCDebug(lcConnman) << "Connected" << *i;
+        Q_EMIT tetheringClientAdded(*i);
```

Comment:
Is there really need to have these separate added and removed signals? On one part they are redundant with the Changed() signal, but also the usage might be error prone when the actual client list updates later and async. I.e. the client app better not fetch the client list upon this and assume it's good.

Reply:
Yes. For showing a notification when a client connects, and disconnects. First added notify = connect, second added notify usually means that IP lease was obtained, it can be requested earlier and this is something we cannot control.